pkg/appendable/io.go

```diff
@@ -145,10 +178,68 @@ func ReadIndexFile(r io.Reader, data DataHandler) (*IndexFile, error) {
 		if _, err := data.Seek(int64(f.EndByteOffsets[len(f.EndByteOffsets)-1]), io.SeekStart); err != nil {
 			return nil, fmt.Errorf("failed to seek data file: %w", err)
 		}
+
+		if _, isCSV := data.(CSVHandler); isCSV {
+			f.EndByteOffsets = f.EndByteOffsets[1:]
+			f.Checksums = f.Checksums[1:]
+		}
```

Comment:
we trim the head off for both `f.EndByteOffsets` and `f.Checksums` because the first element are always headers. 

*Why do we need to store them in the first place?*
The start byte offset must start at the end of the header. So this means we need to store the header information in the byte offset.

We store the header checksum to make the relation between `EndByteOffset` and `Checksums` to be bijective

Reply:
Interesting, I hadn't considered this. There are some implicit assumptions elsewhere in the code that the start byte offset for the first element is always zero, which isn't true in this case.

I think this won't be a problem when we migrate to the B+ tree structure, but just keep that in mind.